cpp/src/copying/gather.cu

```diff
@@ -382,13 +383,15 @@ void gather(table const* source_table, gdf_index_type const gather_map[],
   std::transform(source_table->begin(), source_table->end(),
                  destination_table->begin(), destination_table->begin(),
                  gather_column);
+
 }
 
 }  // namespace detail
 
 void gather(table const* source_table, gdf_index_type const gather_map[],
             table* destination_table) {
   detail::gather(source_table, gather_map, destination_table);
+  nvcategory_gather_table(*source_table, *destination_table);
```

Comment:
I don't think you want to *always* call this do you? Only if it's an NVCATEGORY column and it's not empty, right?

Reply:
That logic is in `nvcategory_gather_table`:

https://github.com/rapidsai/cudf/blob/branch-0.8/cpp/src/string/nvcategory_util.cpp#L41

Reply:
OK.